src/EmotiBitWiFiHost.cpp

```diff
@@ -72,27 +70,166 @@ int8_t EmotiBitWiFiHost::begin()
+		//initially tries broadcasting
+		for (int ip = 0; ip < availableSubnets.size(); ip++) {
+			string braodcastIp = availableSubnets.at(ip) + "." + ofToString(255);
+			ipSearchCxn.Connect(braodcastIp.c_str(), advertisingPort);
+			ipSearchCxn.SetEnableBroadcast(true);
+			ipSearchCxn.SetNonBlocking(true);
+			ipSearchCxn.SetReceiveBufferSize(pow(2, 10));
+			// Send advertising message
+			string packet = EmotiBitPacket::createPacket(EmotiBitPacket::TypeTag::HELLO_EMOTIBIT, advertisingPacketCounter++, "", 0);
+			//ofLogVerbose() << "Sent: " << packet;
+			ipSearchCxn.Send(packet.c_str(), packet.length());
+		}
+	}
+	if(advertisingIps.size() == 0 || !enableBroadcast) { //connects through unicasting
+		if (emotibitSubnets.size() == 0) { //initial search through all subnets
+			getAvailableSubnets(); //if subnet appears after osciliscope is open (mobile hotspot)
+			for (int ip = 0; ip < availableSubnets.size(); ip++) {
+				for (int hostId = 0; hostId < 255; hostId++) {
+					string braodcastIp = availableSubnets.at(ip) + "." + ofToString(hostId);
+					ipSearchCxn.Connect(braodcastIp.c_str(), advertisingPort);
+					ipSearchCxn.SetEnableBroadcast(true);
+					ipSearchCxn.SetNonBlocking(true);
+					ipSearchCxn.SetReceiveBufferSize(pow(2, 10));
+					// Send advertising message
+					string packet = EmotiBitPacket::createPacket(EmotiBitPacket::TypeTag::HELLO_EMOTIBIT, advertisingPacketCounter++, "", 0);
+					//ofLogVerbose() << "Sent: " << packet;
+					ipSearchCxn.Send(packet.c_str(), packet.length());
+				}
+			}
+		}
+		else {
+			for (int hostId = 0; hostId < 255; hostId++) {
+				string braodcastIp = emotibitSubnets.at(0) + "." + ofToString(hostId);
+				ipSearchCxn.Connect(braodcastIp.c_str(), advertisingPort);
+				ipSearchCxn.SetEnableBroadcast(true);
+				ipSearchCxn.SetNonBlocking(true);
+				ipSearchCxn.SetReceiveBufferSize(pow(2, 10));
+				// Send advertising message
+				string packet = EmotiBitPacket::createPacket(EmotiBitPacket::TypeTag::HELLO_EMOTIBIT, advertisingPacketCounter++, "", 0);
+				//ofLogVerbose() << "Sent: " << packet;
+				ipSearchCxn.Send(packet.c_str(), packet.length());
+			}
+		}
+		
+	}
+
+}
+
+void EmotiBitWiFiHost::getAdvertisingIpReceive() {
+	const int maxSize = 32768;
+	if (enableBroadcast) {
+		// Receive advertising messages
+		static char udpMessage[maxSize];
+		int msgSize = ipSearchCxn.Receive(udpMessage, maxSize);
+		if (msgSize > 0)
+		{
+			string rxIp;
+			int rxPort;
+			ipSearchCxn.GetRemoteAddr(rxIp, rxPort);
+			string message = udpMessage;
+			ofLogVerbose() << "Received: " << message;
+
+			vector<string> ipSplit = ofSplitString(rxIp, ".");
+			string subnetAddr = ipSplit.at(0) + "." + ipSplit.at(1) + "." + ipSplit.at(2);
+
+			emotibitSubnets.push_back(subnetAddr);
+			advertisingIps.push_back(subnetAddr + "." + ofToString(255));
+		}
+	}
+	if(advertisingIps.size() == 0 || !enableBroadcast) {
+		for (int msg = 0; msg < 100; msg++) {
```

Comment:
I'm not sure I understand the reason for this `for` loop. Can you please clarify?

Reply:
ofxUDPManager::Receive(...) only returns the first message in the receive buffer, so the for loop is to get all the messsages/emotibits